engine/graphics/src/webgpu/graphics_webgpu_private.h

```diff
@@ -24,218 +24,223 @@
-            m_Sampler             = NULL;
+            m_Type           = TEXTURE_TYPE_2D;
+            m_GraphicsFormat = TEXTURE_FORMAT_RGBA;
+            m_Format         = WGPUTextureFormat_Undefined;
+            m_UsageFlags     = WGPUTextureUsage_None;
+            m_Texture        = NULL;
+            m_TextureView    = NULL;
+            m_Sampler        = NULL;
         }
 
-        WGPUTexture       m_Texture;
-        WGPUTextureView   m_TextureView;
-        WGPUSampler       m_Sampler;
-        TextureType       m_Type;
-        TextureFormat     m_GraphicsFormat;
+        WGPUTexture m_Texture;
+        WGPUTextureView m_TextureView;
+        WGPUSampler m_Sampler;
+        TextureType m_Type;
+        TextureFormat m_GraphicsFormat;
         WGPUTextureFormat m_Format;
         WGPUTextureUsageFlags m_UsageFlags;
-        uint32_t          m_Width;
-        uint32_t          m_Height;
-        uint16_t          m_Depth;
-        uint32_t          m_OriginalWidth;
-        uint32_t          m_OriginalHeight;
-        uint16_t          m_MipMapCount         : 5;
-        uint16_t          m_TextureSamplerIndex : 10;
-        uint32_t          m_Destroyed           : 1;
-        uint32_t          m_UsageHintFlags      : 8;
+        uint32_t m_Width;
+        uint32_t m_Height;
+        uint32_t m_OriginalWidth;
+        uint32_t m_OriginalHeight;
+        uint16_t m_Depth;
+        uint16_t m_TextureSamplerIndex : 10;
+        uint16_t m_MipMapCount : 5;
+        uint8_t m_UsageHintFlags;
+        uint8_t m_Destroyed : 1;
+        uint8_t : 8; // aligned by 32 bit
     };
 
     struct WebGPURenderTarget
     {
-        WebGPURenderTarget() {
+        WebGPURenderTarget()
+        {
             memset(this, 0, sizeof(*this));
         }
 
-        AttachmentOp   m_ColorBufferLoadOps[MAX_BUFFER_COLOR_ATTACHMENTS];
-        AttachmentOp   m_ColorBufferStoreOps[MAX_BUFFER_COLOR_ATTACHMENTS];
-        float          m_ColorBufferClearValue[MAX_BUFFER_COLOR_ATTACHMENTS][4];
+        AttachmentOp m_ColorBufferLoadOps[MAX_BUFFER_COLOR_ATTACHMENTS];
+        AttachmentOp m_ColorBufferStoreOps[MAX_BUFFER_COLOR_ATTACHMENTS];
+        float m_ColorBufferClearValue[MAX_BUFFER_COLOR_ATTACHMENTS][4];
 
-        HTexture       m_TextureResolve[MAX_BUFFER_COLOR_ATTACHMENTS];
-        HTexture       m_TextureColor[MAX_BUFFER_COLOR_ATTACHMENTS];
-        HTexture       m_TextureDepthStencil;
+        HTexture m_TextureResolve[MAX_BUFFER_COLOR_ATTACHMENTS];
+        HTexture m_TextureColor[MAX_BUFFER_COLOR_ATTACHMENTS];
+        HTexture m_TextureDepthStencil;
 
-        uint32_t       m_Width;
-        uint32_t       m_Height;
-        float          m_Scissor[4];
-        uint8_t        m_Multisample;
-        uint32_t       m_ColorBufferCount : 7;
+        float m_Scissor[4];
+        uint32_t m_Width;
+        uint32_t m_Height;
+        uint8_t m_Multisample;
+        uint32_t m_ColorBufferCount : 7;
+        uint32_t : 17;
```

Comment:
Try align structure.

Reply:
I'm assuming you're after struct size, and not alignment?
We have separate defines for alignment. (DM_ALIGN)

Also, this change doesn't affect size of the struct, it's still 194 bytes.
(I tried to see if I could get it smaller, and while I got it down to 190 bytes, the readability also suffered)

The extra implicit padding doesn't do anything to the size.
We might change the uint32 to an uint8 though, but that's not any real gain.

Reply:
Removed.